src/components/templates/PartyMap/PartyMap.tsx

```diff
@@ -29,37 +30,51 @@ export const PartyMap: React.FC<PartyMapProps> = ({ venue }) => {
   const { user, profile } = useUser();
   const { recentVenueUsers } = useRecentVenueUsers({ venueName: venue.name });
 
-  const { relatedVenues } = useRelatedVenues({ currentVenueId: venue.id });
-
-  const selfAndChildVenueIds = useMemo(
-    () =>
-      relatedVenues
-        .filter(
-          (relatedVenue) =>
-            relatedVenue.parentId === venue.id || relatedVenue.id === venue.id
-        )
-        .map((childVenue) => childVenue.id),
-    [relatedVenues, venue]
+  const [selectedRoom, setSelectedRoom] = useState<Room | undefined>();
+
+  const hasSelectedRoom = !!selectedRoom;
+
+  const [portalVenueId] = useMemo(
+    () => getLastUrlParam(selectedRoom?.url ?? ""),
+    [selectedRoom]
   );
 
-  const { events: selfAndChildVenueEvents } = useVenueEvents({
-    venueIds: selfAndChildVenueIds,
+  const venueId = useMemo(() => (hasSelectedRoom ? portalVenueId : venue.id), [
+    venue,
+    hasSelectedRoom,
+    portalVenueId,
+  ]);
```

Comment:
no need for useMemo here

Reply:
fixed in 099394fba9857a6db1e3c32e2607bbe27f4118f7